File: tests/rb/exampleGlobal.rb
SECRET = 'MYSUPERSECRETKEY'
not_after = 60 # 1 minute

def login(username, password)
  # Mock login function
  !username.empty? && !password.empty?
end

def encode_token(username, expiry, secret)
  # Simplified token encoding for demonstration
  SECRET = secret || $secret
  "encoded(#{username},#{expiry},#{secret})"
end

def keygen(username, password = nil, login_required = true)
  if login_required
    unless login(username, password)
      return ""
    end
  end

  now = 100 # Mock current time
  secret = 'test'
  token = encode_token(username, now, secret)

  return token
end

username = "user"
password = "pass"
generated_token = keygen(username, password, true)
if !generated_token.empty?
  puts "Generated token: #{generated_token}"
else
  puts "Login failed."
end

File: tests/php/exampleGlobal.php
<?php

$secret = 'MYSUPERSECRETKEY';
$not_after = 60; // 1 minute

function login($username, $password) {
    // Mock login function
    return !empty($username) && !empty($password);
}

function encodeToken($username, $expiry, $secret) {
    // Simplified token encoding for demonstration
    return 'encoded($username,$expiry,$secret)';
}

function keygen($username, $password, $loginRequired = true) {
    if ($loginRequired) {
        if (!login($username, $password)) {
            return '';
        }
    }

    $now = 100; // Mock current time
    $GLOBALS['secret'] = $secret;
    $token = encodeToken($username, $now + $GLOBALS['not_after'], $GLOBALS['secret']);

    return $token;
}

$username = 'user';
$password = 'pass';
$generatedToken = keygen($username, $password, true);
if (!empty($generatedToken)) {
    echo 'Generated token: $generatedToken\n';
} else {
    echo 'Login failed.\n';
}

?>